ecms_base/modules/custom/ecms_languages/src/LanguageNegotiationSessionFix.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_languages;
+
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationSession;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Alter the session language negotiator to add query params on node operations.
+ *
+ * @package Drupal\ecms_languages
+ */
+class LanguageNegotiationSessionFix extends LanguageNegotiationSession {
+
+  /**
+   * Node operations to add the language query parameter.
+   */
+  const NODE_OPERATIONS = [
+    'edit',
+    'delete',
+  ];
+
+  /**
+   * {@inheritDoc}
+   */
+  public function processOutbound($path, &$options = [], Request $request = NULL, BubbleableMetadata $bubbleable_metadata = NULL): string {
+    $parent = parent::processOutbound($path, $options, $request, $bubbleable_metadata);
+
+    $parts = explode('/', ltrim($path, '/'));
+
+    // Make sure we have the correct path length.
+    if (count($parts) !== 3) {
+      return $parent;
+    }
+
+    // Make sure we are dealing with a node.
+    if ($parts[0] !== 'node') {
+      return $parent;
+    }
```

Comment:
Will we need this for term translation possibly?

Reply:
Since we're trying to rush this into a deployment, I think we should add that thought to the backlog. Thoughts?

Reply:
Totally fine!